### ISSUE TITLE
Incorrect values within `molecule_atom_indices`

### ISSUE THREAD

**Issue by amorehead** [lucidrains/alphafold3-pytorch#187]
For example, in the following code,

https://github.com/lucidrains/alphafold3-pytorch/blob/8196931836cee2b9f25628fbd3d1c317e2758fb8/alphafold3_pytorch/alphafold3.py#L3599

I believe `molecule_atom_indices` are being used incorrectly with `torch.gather()` currently, because `molecule_atom_indices` contains a single valid (non-negative and non-zero) value currently only for one atom in each `Chem.Mol` object, not for each token. I believe we are confusing molecule-level and token-level indices in `molecule_atom_indices` currently, because I believe `molecule_atom_indices` should really be called `token_atom_indices` which contains a single unique index for each "token center atom" which can be used with `torch.gather()` to pool from atom-level features to *token* (not molecule)-level features. Currently, `molecule_atom_indices` will have `torch.gather()` give you back a correctly-shaped token-sized tensor, but the end values of this tensor are clearly not correct since they represent many duplicated values based on the padding value (index `0).

Fortunately, `molecule_atom_indices` is only being used for confidence head predictions and labels, but we should consider refactoring it into `token_atom_indices` since I believe that is what we are really interested in.

**Comment by lucidrains** [comment]
@amorehead yes indeed, and all this confusion stems from the edge cases of ligands and modified biomolecules being one atom per token

think the proper way to migrate this is to first rename `molecule` to `token` as you said, including renaming this dimension to `token` and all `m` -> `t`. then during input construction, we will need to construct and pass in the actual `molecule_atom_indices`

**Comment by amorehead** [comment]
[This PR](https://github.com/lucidrains/alphafold3-pytorch/pull/188) should fix the training-time distogram and molecule atom indices for ligands and modified polymer residues s.t. distogram and confidence head predictions are correctly supervised for these molecule types.
